api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3895,6 +3956,247 @@ public ObjectLockConfiguration getDefaultRetention(String bucketName)
+    throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+           InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+           InternalException, InvalidResponseException, InvalidArgumentException {
+
+    if (config == null ) {
+      throw new InvalidArgumentException("null value is not allowed in config.");
+    }
+
+    Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("retention", "");
+
+    if (versionId == null) {
+      queryParamMap.put("versionId", "");
+    } else {
+      queryParamMap.put("versionId", versionId);
+    }
+
+    Map<String, String> headerMap = new HashMap<>();
+    if (bypassGovernanceRetention) {
+      headerMap.put("x-amz-bypass-governance-retention", "True");
+    }
+
+    HttpResponse response = executePut(bucketName, objectName, headerMap, queryParamMap, config, 0);
+    response.body().close();
+  }
+
+  /**
+   * Fetches object retention lock of an object.
+   *
+   * </p><b>Example:</b><br>
+   * <pre>{@code
+   * ObjectRetentionConfiguration objectRetentionConfiguration = minioClient.getObjectRetention("my-bucketname", 
+   * "my-object", "version-Id" );
+   * System.out.println("Mode " + objectRetentionConfiguration.mode()); 
+   * System.out.println("Retanetion Until  " + objectRetentionConfiguration.getRetentionDate()); }</pre>
+   *
+   * @param bucketName Bucket name.
+   * @param objectName Object name.
+   * @param versionId  Version Id.
+   *
+   * @throws InvalidBucketNameException  upon invalid bucket name is given
+   * @throws NoSuchAlgorithmException
+   *           upon requested algorithm was not found during signature calculation
+   * @throws InsufficientDataException  upon getting EOFException while reading given
+   *           InputStream even before reading given length
+   * @throws IOException                 upon connection error
+   * @throws InvalidKeyException
+   *           upon an invalid access key or secret key
+   * @throws NoResponseException         upon no response from server
+   * @throws XmlPullParserException      upon parsing response xml
+   * @throws ErrorResponseException      upon unsuccessful execution
+   * @throws InternalException           upon internal library error
+   * @throws InvalidResponseException    upon a non-xml response from server
+   */
+  public ObjectRetentionConfiguration getObjectRetention(String bucketName, String objectName, String versionId)
+    throws InvalidBucketNameException, NoSuchAlgorithmException, InsufficientDataException, IOException,
+           InvalidKeyException, NoResponseException, XmlPullParserException, ErrorResponseException,
+           InternalException, InvalidResponseException {
+
+    Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("retention", "");
+
+    if ( !(versionId == null || versionId.isEmpty())) {
+      queryParamMap.put("versionId",versionId);
+    }
```

Comment:
I don't know why you are missing out of `versionId` handling. Are you deferring in other places?